File: src/main/java/io/github/mdsimmo/bomberman/events/BmIntentCancellable.java
package io.github.mdsimmo.bomberman.events;

import org.bukkit.event.Cancellable;

public abstract class BmIntentCancellable extends BmIntent implements Cancellable {

    private boolean cancelled;

    @Override
    public void verifyHandled() {
        if (!isHandled() && !isCancelled())
            throw new RuntimeException("Event not handled: " + this);
    }

    @Override
    public boolean isCancelled() {
        return cancelled;
    }

    @Override
    public void setCancelled(boolean cancel) {
        this.cancelled = cancel;
    }
}

File: src/main/java/io/github/mdsimmo/bomberman/events/BmIntentCancellable.kt
package io.github.mdsimmo.bomberman.events

class BmIntentCancellable : IntentCancellable {
    private var cancelled = false
    private var handled = false

    override fun verifyHandled() {
        if (!handled && !isCancelled) throw RuntimeException("Event not handled: $this")
    }

    override fun isCancelled(): Boolean {
        return cancelled
    }

    override fun setCancelled(cancel: Boolean) {
        cancelled = cancel
    }

    override fun isHandled(): Boolean = handled

    override fun setHandled() {
        handled = true
    }
}